### DISCUSSION TITLE
Change language to spanish

### DISCUSSION THREAD

**Discussion by Narendra678** [apache/superset#18806] [Q&A / Help]
Hi Team,

Can i change language to spanish?

![image](https://user-images.githubusercontent.com/88739186/154542898-1db683c1-e440-4363-ba83-e963ff18ee35.png)

Regards,
Naren

**Comment by srinify** [comment]
Hi @Narendra678 Github issues is for reporting of bugs or occasional feature requests / design conversations. I'm moving this over to Github Discussions where someone can try to help (or you can also join the Superset Slack)

**Comment by 1c7** [comment]
# Tutorial: How to enable Superset language selection

## Background Info: as of 2022-Feb-17, Superset version 1.4.1
Language Selection are disable by default, you can see that in file `superset/config.py`  

<img width="1298" alt="image" src="https://user-images.githubusercontent.com/1804755/154639338-0654fbfa-5101-41c1-99b7-bca55cd2a5ef.png">

## So, How to enable that?

## Step 1: Create a `superset_config.py` file
in my case I need to use Chinese: 
```
LANGUAGES = {
    'en': {'flag': 'us', 'name': 'English'},
    "zh": {"flag": "cn", "name": "Chinese"},
}
```

in your case you need Spanish
```
LANGUAGES = {
    'en': {'flag': 'us', 'name': 'English'},
    "es": {"flag": "es", "name": "Spanish"},
}
```

Important: You have to config **AT LEAST** 2 Language to enable language selection

## Step 2: add `superset_config.py` to proper PYTHONPATH location
Like this:
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/1804755/154639567-ca1d7b4d-1bf8-4c4c-ab1b-c44adeb327e7.png">

This line put `superset_config.py` into proper location.
```
COPY superset_config.py /app/pythonpath/superset_config.py
```

Why `/app/pythonpath/`? 
because in Docker Image `apache/superset`, it set default `PYTHONPATH` to `/app/pythonpath/`     

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/1804755/154639744-1cfe3a59-4181-4686-ac50-327f0b4de5ca.png">

# The result look like this

<img width="390" alt="image" src="https://user-images.githubusercontent.com/1804755/154639906-4e62dc5d-c79f-4e79-8a37-c07c7cfac6ac.png">

**Reply by Narendra678** [reply]
Hi Sir,

Thanks much for your detailed explanation. Can we make it to only one specific report? I mean only one report should show spanish rest should be english.

And can we hide 'view query' option ?


Regards,
Naren

**Reply by 1c7** [reply]
Hi Naren,
I don't have experience with Superset reports.
no idea how to do that.

**Reply by Narendra678** [reply]
Hi Sir,

It is working but not converting all the values.
![image](https://user-images.githubusercontent.com/88739186/155497794-accb241f-1db4-43a4-b7bb-28de24e445dc.png)

Can you suggest here ? please

Regards,
Naren

**Reply by 1c7** [reply]
The language section feature is disabled by default for a reason.
most languages are not fully translated yet.

You can consider contributing some Spanish translation into Superset source code

**Reply by han916701843** [reply]
> Please tell me,what's the version of superset image(the version of my superset image is master)?when I complete config follow your step,it doesn't work(there isn't any lanuage section),Thank you!

**Reply by 1c7** [reply]
Superset version 1.4.2 should work fine.

**Comment by Narendra678** [comment]
Hi Sir,

I did not understand this "You can consider contributing some Spanish translation into Superset source code"

you mean we can not do this?

Regards,
Naren

**Reply by 1c7** [reply]
You can fork the Superset source code repo on Github.
Modify Spanish translation and send Pull Request.

**Reply by lvazquez** [reply]
I have a complete Spanish translation of messages.po for Superset 2.0.0 I have made for a personal Superset based project and I would like to contribute this, but I don't have a lot of spare time. What would be the easier way to contribute with this?

**Reply by torzsmokus** [reply]
@lvazquez creating a pull request with your changes takes not much more time than writing this comment, I hope you had that time since

**Comment by Narendra678** [comment]
Do i need to modify below code?

LANGUAGES = {
    'en': {'flag': 'us', 'name': 'English'},
    "es": {"flag": "es", "name": "Spanish"},
}

**Reply by 1c7** [reply]
it seem fine to me

**Comment by qishibo** [comment]
@1c7 after adding `superset_config.py` to `/app/pythonpath/` by `docker exec ...`
I can't login with default `admin` `admin` , do you know why this?

**Reply by 1c7** [reply]
have you run 
```
docker exec -it superset superset fab create-admin \
              --username admin \
              --firstname Superset \
              --lastname Admin \
              --email admin@superset.com \
              --password admin
```
like the docs says?
https://hub.docker.com/r/apache/superset

**Comment by dvconsultores** [comment]
How to contribute with translate?

**Reply by torzsmokus** [reply]
Open https://github.com/apache/superset/blob/master/superset/translations/es/LC_MESSAGES/messages.po for Spanish (or the same file corresponding to your language) and click Edit (pencil icon top right). That will fork the project for you, and when you are ready with the updates, create a pull request.
